src/core/applicative_error.rs

```diff
@@ -0,0 +1,69 @@
+use super::{
+    applicative::ApplicativeInstance,
+    prelude::{ApplyInstance, FunctorInstance},
+};
+
+pub trait ApplicativeErrorTy {
+    type Cons<T>: ApplicativeErrorInstance<T, Kind = Self>
+        + ApplicativeInstance<Kind = Self>
+        + ApplyInstance<T, Kind = Self>
+        + FunctorInstance<T, Kind = Self>;
```

Comment:
I wonder if this ends up being redundant and just having `Cons<T>: ItselfInstance<...> + DirectParentsInstances<...>` would be enough

For this case maybe it would be correct and enough to just say

```rust
pub trait ApplicativeErrorTy {
    type Cons<T>: ApplicativeErrorInstance<T, Kind = Self> 
        + ApplicativeInstance<Kind = Self>;
}
```